src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -42,572 +42,426 @@ export interface SearchBarInjectedDeps {
-          data-test-subj="unifiedFilterBar"
-        />
-      );
-    }
-
-    return (
-      <div className={classes} css={cssStyles} data-test-subj="globalQueryBar">
-        <QueryBarTopRow<QT>
-          timeHistory={this.props.timeHistory}
-          query={this.state.query}
-          screenTitle={this.props.screenTitle}
-          onSubmit={this.onQueryBarSubmit}
-          indexPatterns={this.props.indexPatterns}
-          isLoading={this.props.isLoading}
-          fillSubmitButton={this.props.fillSubmitButton || false}
-          prepend={this.props.showFilterBar || this.props.showQueryInput ? queryBarMenu : undefined}
-          showDatePicker={this.props.showDatePicker}
-          dateRangeFrom={this.state.dateRangeFrom}
-          dateRangeTo={this.state.dateRangeTo}
-          isRefreshPaused={this.props.isRefreshPaused}
-          refreshInterval={this.props.refreshInterval}
-          showAutoRefreshOnly={this.props.showAutoRefreshOnly}
-          showQueryInput={this.props.showQueryInput}
-          showAddFilter={this.props.showFilterBar}
-          isDisabled={this.props.isDisabled}
-          onRefresh={this.props.onRefresh}
-          onRefreshChange={this.props.onRefreshChange}
-          onChange={this.onQueryBarChange}
-          isDirty={this.isDirty()}
-          customSubmitButton={
-            this.props.customSubmitButton ? this.props.customSubmitButton : undefined
-          }
-          showSubmitButton={this.props.showSubmitButton}
-          submitButtonStyle={this.props.submitButtonStyle}
-          dataTestSubj={this.props.dataTestSubj}
-          indicateNoData={this.props.indicateNoData}
-          placeholder={this.props.placeholder}
-          isClearable={this.props.isClearable}
-          iconType={this.props.iconType}
-          nonKqlMode={this.props.nonKqlMode}
-          timeRangeForSuggestionsOverride={timeRangeForSuggestionsOverride}
-          filters={this.props.filters!}
-          onFiltersUpdated={this.props.onFiltersUpdated}
-          dataViewPickerComponentProps={this.props.dataViewPickerComponentProps}
-          textBasedLanguageModeErrors={this.props.textBasedLanguageModeErrors}
-          onTextBasedSavedAndExit={this.props.onTextBasedSavedAndExit}
-          showDatePickerAsBadge={this.shouldShowDatePickerAsBadge()}
-          filterBar={filterBar}
-          suggestionsSize={this.props.suggestionsSize}
-          isScreenshotMode={this.props.isScreenshotMode}
-          onTextLangQuerySubmit={this.onTextLangQuerySubmit}
-          onTextLangQueryChange={this.onTextLangQueryChange}
-        />
-      </div>
-    );
-  }
-
-  private hasFiltersOrQuery() {
-    const hasFilters = Boolean(this.props.filters && this.props.filters.length > 0);
-    const hasQuery = Boolean(
-      this.state.query && isOfQueryType(this.state.query) && this.state.query.query
-    );
-    return hasFilters || hasQuery;
-  }
-
-  private renderSavedQueryManagement = memoizeOne(
```

Comment:
💬 Broke this up into its own component which is memoized with `React.memo` instead.